src/main/java/com/caedis/duradisplay/config/DuraDisplayConfig.java

```diff
@@ -46,107 +45,20 @@ public static void reloadConfigObject() {
-
-        ChargeConfig.ShowWhenFull = config.getBoolean(
-            "ShowWhenFull",
-            DuraDisplayConfig.CATEGORY_CHARGE,
-            ChargeConfig.ShowWhenFull,
-            "Show charge percentage when item is full");
-
-        DurabilityConfig.UseColorThreshold = config.getBoolean(
-            "UseColorThreshold",
-            DuraDisplayConfig.CATEGORY_DURABILITY,
-            DurabilityConfig.UseColorThreshold,
-            "Use the color thresholds instead of the default gradient coloring.");
-
-        ChargeConfig.UseColorThreshold = config.getBoolean(
-            "UseColorThreshold",
-            DuraDisplayConfig.CATEGORY_CHARGE,
-            ChargeConfig.UseColorThreshold,
-            "Use the color thresholds instead of the static blue color.");
-
-        Property dura_colorThresh = config.get(
-            CATEGORY_DURABILITY,
-            "ColorThresholds",
-            DurabilityConfig.ColorThreshold,
-            "List of numbers in ascending order from 0-100 that set the thresholds for durability color mapping. "
-                + "Colors are from Red -> Yellow -> Green with Red being less than or equal to the first value "
-                + "and Green being greater than or equal to the last value");
-        DurabilityConfig.ColorThreshold = dura_colorThresh.getDoubleList();
-
-        // clean up whatever the user inputs
-        DurabilityConfig.ColorThreshold = Arrays.stream(DurabilityConfig.ColorThreshold)
-            .filter(num -> num >= 0.0 && num <= 100.0)
-            .sorted()
-            .toArray();
-        dura_colorThresh.set(DurabilityConfig.ColorThreshold);
-
-        Property charge_colorThresh = config.get(
-            CATEGORY_CHARGE,
-            "ColorThresholds",
-            ChargeConfig.ColorThreshold,
-            "List of numbers in ascending order from 0-100 that set the thresholds for charge color mapping. "
-                + "Colors are from Red -> Orange -> Blue with Red being less than or equal to the first value "
-                + "and Blue being greater than or equal to the last value");
-        ChargeConfig.ColorThreshold = charge_colorThresh.getDoubleList();
-
-        // clean up whatever the user inputs
-        ChargeConfig.ColorThreshold = Arrays.stream(ChargeConfig.ColorThreshold)
-            .filter(num -> num >= 0.0 && num <= 100.0)
-            .sorted()
-            .toArray();
-        charge_colorThresh.set(ChargeConfig.ColorThreshold);
+        for (var c : ConfigInfo.getConfigs()) {
+            c.loadConfig(config);
+        }
 
         if (config.hasChanged()) {
             config.save();
         }
 
         // Gregtech Bars
-        GT_Mod.gregtechproxy.mRenderItemDurabilityBar = Enable
-            && !(DurabilityConfig.Enabled && !DurabilityConfig.RenderBar);
-        GT_Mod.gregtechproxy.mRenderItemChargeBar = Enable && !(ChargeConfig.Enabled && !ChargeConfig.RenderBar);
+        GT_Mod.gregtechproxy.mRenderItemDurabilityBar = false;
```

Comment:
I see you fell for the refactor suggestion.

This logic is actually needed as otherwise the default bars for these 2 mods would never show.

Reply:
[95e6ca3](https://github.com/GTNewHorizons/DuraDisplay/pull/9/commits/95e6ca311e0cffc6ed781ae0f076723735d4c904)